File: java/solucion.java
public class SumaPrimos {
    public static void main(String[] args) {
        long inicio = System.currentTimeMillis();
        long suma = 0;
        int contador = 0;
        for (long i = 2; contador < 900000; i++) {
            boolean primo = true;
            for (long j = 2; j * j <= i; j++) {
                if (i % j == 0) {
                    primo = false;
                    break;
                }
            }
            if (primo) {
                suma += i;
                contador++;
            }
        }

        boolean sumaEsPrimo = true;
        for (long i = 2; i * i <= suma; i++) {
            if (suma % i == 0) {
                sumaEsPrimo = false;
                break;
            }
        }

        long fin = System.currentTimeMillis();
        System.out.println("La suma de los primeros 900,000 primos es: " + suma);
        System.out.println("La suma es prima: " + (sumaEsPrimo ? "Si" : "No"));
        System.out.println("Tiempo: " + (fin - inicio) / 1000.0 + " segundos");
    }
}

File: python/solucion.py
import time

def es_primo(n):
    if n <= 1:
        return False
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            return False
    return True

inicio = time.time()
suma = 0
contador = 0
n = 2
while contador < 900000:
    if es_primo(n):
        suma += n
        contador += 1
    n += 1

suma_es_primo = es_primo(suma)
fin = time.time()

print(f"La suma de los primeros 900,000 primos es: {suma}")
print(f"La suma es prima: {'Si' if suma_es_primo else 'No'}")
print(f"Tiempo: {fin - inicio:.2f} segundos")
